### ISSUE TITLE
Reader seems to accumulate annotations between invocations

### ISSUE THREAD

**Issue by therapon** [amzn/ion-js#61]
Annotations should **not** accumulate  between **new** instances of IonReader. 

```
r = require("ion-js").makeReader("ann1a::ann1b::10 ann2::11"); r.next(); r.annotations()
[ 'ann1a', 'ann1b' ]
> r = require("ion-js").makeReader("ann1a::ann1b::10 ann2::11"); r.next(); r.annotations()
[ 'ann1a', 'ann1b', 'ann1a', 'ann1b' ]
> r = require("ion-js").makeReader("ann1a::ann1b::10 ann2::11"); r.next(); r.annotations()
[ 'ann1a', 'ann1b', 'ann1a', 'ann1b', 'ann1a', 'ann1b' ]
```

Should verify reader's beahviour when for the same reader instance we navigate to multiple values that have annotations. The API should return the annotations of the current value being pointed to by the reader.

**Comment by mtcalvin** [comment]
This also seems to be masking (via causing the test suite to never reach cases where the first annotation is `$ion_symbol_table`) bugs in the symbol table parsing.

**Comment by wesboyt** [comment]
This has been solved on PR #118
